File: flows/flows/report/bill_tracking/bill_tracking.py
# Copyright (c) 2013, Arun Logistics and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe


def execute(filters=None):
	columns, data = get_columns(filters), get_date(filters)
	return columns, data


def get_columns(filters):
	return [
		"Indent Date:Date:100",
		"Indent:Link/Indent:100",
		"Customer::250",
		"Bill Date:Date:100",
		"Bill No.::100",
		"Bill Receive Date:Date:100"
	]


def get_date(filters):
	return frappe.db.sql("""
	SELECT indent.posting_date, indent.name, indent_item.customer, indent_invoice.transaction_date,
	indent_invoice.invoice_number, ifnull(indent_invoice.invoice_receive_date, '')
	FROM `tabIndent Item` indent_item, `tabIndent` indent, `tabIndent Invoice` indent_invoice
	WHERE indent_item.parent = indent.name
	AND indent_item.name = indent_invoice.indent_item
	AND indent_invoice.docstatus != 2
	AND indent.posting_date BETWEEN '{from_date}' AND '{to_date}'
	ORDER BY indent.posting_date desc;
	""".format(**filters))

File: flows/flows/report/bill_tracking/bill_tracking.js
// Copyright (c) 2013, Arun Logistics and contributors
// For license information, please see license.txt

var el = document.createElement('script');
var att = document.createAttribute("src");
att.value = "http://mleibman.github.io/SlickGrid/slick.editors.js";
el.setAttributeNode(att);

document.getElementsByTagName('head')[0].appendChild(el);

frappe.query_reports["Bill Tracking"] = {
	"filters":[
		{
			"fieldname":"from_date",
			"label":__("From Date"),
			"fieldtype":"Date",
			"default":frappe.datetime.month_start(),
			"reqd":1
		},
		{
			"fieldname":"to_date",
			"label":__("To Date"),
			"fieldtype":"Date",
			"default":frappe.datetime.month_end(),
			"reqd":1
		}
	],
	startup:function (grid, dataview) {
		grid.onCellChange.subscribe(function (e, args) {
			grid.invalidateRow(args.row);
			var row = dataview.getItem(args.row);
			//row[grid.getColumns()[args.cell].field] = 'Yay';

			if (!grid.getColumns()['change']) row[grid.getColumns()['change']] = [];

			grid.getColumns()['change'].push(grid.getColumns()[args.cell].field);
			dataview.updateItem(args.item.id, row);
			grid.render();
		});
	}
};

var report = frappe.query_report;
report.appframe.add_button(__('Update'), function () {
	console.log("Update Triggered");
}, "icon-thumbs-up");

frappe.query_report.slickgrid_options["editable"] = true;
frappe.query_report.slickgrid_options["autoEdit"] = false;